### ISSUE TITLE
Serious Sam 2017 Editor (564380)

### ISSUE THREAD

**Issue by kodatarule** [ValveSoftware/Proton#4496]
# Compatibility Report
- Name of the game with compatibility issues:Serious Sam 2017 Editor (Tool)
- Steam AppID of the game:564380

## System Information
- GPU: Geforce RTX 3070
- Driver/LLVM version: nvidia 460.27.04
- Kernel version: 5.9.14-zen1-1-zen
- Link to full system information report as [Gist](https://gist.github.com/):https://gist.github.com/kodatarule/69826461fe7f43ab0a799d82b0b7b919
- Proton version: Proton Experimental

## I confirm:
- [x] that I haven't found an existing compatibility report for this game.
- [x] that I have checked whether there are updates for my system available.

[steam-564380.log](https://github.com/ValveSoftware/Proton/files/5742009/steam-564380.log)

## Symptoms:
![Screenshot_20201225_151555](https://user-images.githubusercontent.com/65848953/103136182-e090ff80-46c6-11eb-9eee-dc8a70b10c43.png)
![Screenshot_20201225_151538](https://user-images.githubusercontent.com/65848953/103136183-e25ac300-46c6-11eb-87e6-7c8906529472.png)

 Gray screen

## Reproduction

Simply launch the editor tool with Proton Experimental(if you try any other version it would give out access violation so there is something in proton experimental that no other proton has). The app will launch just fine and if you click top left for file or any other you'll see that it's indeed working and functional you can even load a level just fine and start a simulation, but it's just that the GUI interface is gray screen.

**Comment by kisak-valve** [comment]
Hello @kodatarule, `fixme:vulkan:X11DRV_vkCreateWin32SurfaceKHR Application requires child window rendering, which is not implemented yet!` in the log points towards https://bugs.winehq.org/show_bug.cgi?id=45277.

**Comment by kodatarule** [comment]
I tried with Proton GE even when it is implemented it just says access violation I can give 2nd log of that if you are interested and thanks for the fast reply.

**Comment by F1xteR** [comment]
> I tried with Proton GE even when it is implemented it just says access violation I can give 2nd log of that if you are interested and thanks for the fast reply.

Well it would be helpful if you send the 2nd log

**Comment by kodatarule** [comment]
Yeah that makes sense :D well here it is this is the log when I use Proton 5.21 GE 1. The app just spits error The application malfunctioned and it will now close and the log indicates for some access violation.
[steam-564380.log](https://github.com/ValveSoftware/Proton/files/5743587/steam-564380.log)

**Comment by kodatarule** [comment]
Upon further testing I changed the renderer to the native vulkan using proton experimental and then I switched to proton 6.0rc4 tkg that uses the system wide wine tkg 6.0rc4 and it just auto closed. Here is the log and I think the error it points to is this:
X Error of failed request:  BadWindow (invalid Window parameter)
  Major opcode of failed request:  12 (X_ConfigureWindow)
  Resource id in failed request:  0x64000ab
  Serial number of failed request:  370
  Current serial number in output stream:  373
[steam-564380.log](https://github.com/ValveSoftware/Proton/files/5746526/steam-564380.log)

![Screenshot_20201228_105802](https://user-images.githubusercontent.com/65848953/103202720-ddcb2000-48fb-11eb-9eba-edb494ac5aac.png)
![Screenshot_20201228_105816](https://user-images.githubusercontent.com/65848953/103202724-de63b680-48fb-11eb-8572-914f16937474.png)

**Comment by kodatarule** [comment]
UPDATE:
After the latest update to proton-experimental that includes different child window rendering patch, both serious sam 2017 editor and the newly released serious sam 2021 editor(for serious sam 4) are working!(Albeit a little bit on the fast side, but that's another issue that is already reported.
![Screenshot_20210402_000558](https://user-images.githubusercontent.com/65848953/113389191-3ea14300-9398-11eb-9946-8c7acab4aa8b.png)
![Screenshot_20210402_000803](https://user-images.githubusercontent.com/65848953/113389194-406b0680-9398-11eb-8a12-84a800ac743e.png)
![Screenshot_20210402_002216](https://user-images.githubusercontent.com/65848953/113389199-4234ca00-9398-11eb-9050-a911716125e2.png)

**Comment by DTHQDTC** [comment]
Ooooooooooooooooooooo-kay... So Serious Sam 2017 Editor is gone?
